app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/LibraryFragment.java

```diff
@@ -31,20 +32,19 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         View root = inflater.inflate(R.layout.fragment_library, container, false);
         final FloatingActionButton addBookButton = root.findViewById(R.id.add_button);
 
-        addBookButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                openAddEditActivity();
-            }
-        });
+        addBookButton.setOnClickListener(
+                view -> openAddEditActivity()
+        );
         return root;
     }
 
     /**
      * function allows user to jump into the add/edit screen when click on the floating button
      */
     private void openAddEditActivity() {
+        // TODO: Change the enum when calling the activity for editing
         Intent intent = new Intent(this.getActivity(), AddEditActivity.class);
+        intent.putExtra("BOOK_ACTION", BOOK_ACTION.ADD);
```

Comment:
I think a good practice here is to declare the key "BOOK_ACTION" as static final. Something like this:
public static final String BOOK_ACTION_KEY = "com.cmput301f20t21.bookfriends.BOOK_ACTION";
And then in AddEditActivity.java:
action = (BOOK_ACTION) intent.getSerializableExtra(LibraryFragment.BOOK_ACTION_KEY);

https://stackoverflow.com/questions/57010358/best-practice-for-name-putextra/57010913